README.md

```diff
@@ -189,6 +189,24 @@ $prunedHtml = HtmlPruner::fromDomDocument($cssInliner->getDomDocument())
   ->removeRedundantClassesAfterCssInlined($cssInliner)->render();
 ```
 
+The `removeElementsWithDisplayNone` method will not remove any elements which
+have the class `-emogrifier-keep`.  So if, for example, there are elements which
+by default have `display: none` but are revealed by an `@media` rule, or which
+are intended as a preheader, you can add that class to those elements.  The
+paragraph in this HTML snippet will not be removed even though it has
+`display: none` (which has presumably been applied by `CssInliner::inlineCss()`
+from a CSS rule `.preheader { display: none; }`):
+
+```html
+<p class="preheader -emogrifier-keep" style="display: none;">
```

Comment:
I'm wondering: Why the leading hyphen in `-emogrifier-keep`?

Reply:
This is to give it a ‘vendor prefix’ as outlined in the [W3C](https://www.w3.org/TR/CSS/#future-proofing) [specification](https://www.w3.org/TR/CSS2/syndata.html#vendor-keywords).

When parsing the CSS, etc., Emogrifier is acting very much as a user agent, and following requirements like “user agents must …” and “user agents should …”.  Although the specification generally refers to using vendor prefixes for keywords and property names, it seems appropriate to extend the principle to proprietary class names.  The leading hyphen should ensure there will not be a clash with any user-defined class names.

Reply:
Ah, makes sense. Thanks for the explanation.